pandas/core/generic.py

```diff
@@ -8990,13 +8991,21 @@ def compare(
             else:
                 self = self[mask]
                 other = other[mask]
+        if not isinstance(suffixes, tuple):
```

Comment:
Why not raising immediately? This is a new argument, so we can be as strict as we want

Reply:
I edited it.
What do you think are other limitations on the input that can make it invalid?

Reply:
@phofl I was thinking one test maybe the type of the elements inside the tuple. Do we only limit the type to string? float and other numeric values work too, do we include them?

Reply:
We do at other places, so would be ok I think

Reply:
Ok cool. So is it done?